app/code/Magento/Inventory/Model/Api/SearchCriteria/CollectionProcessor/FulltextFilterProcessor/FulltextFilter.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\Api\SearchCriteria\CollectionProcessor\FulltextFilterProcessor;
+
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor\CustomFilterInterface;
+use Magento\Framework\Data\Collection\AbstractDb;
+use Magento\Framework\Api\Filter;
+use Magento\Framework\DB\Select;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Class FulltextFilter
+ */
+class FulltextFilter implements CustomFilterInterface
+{
+    /**
+     * Returns list of columns from fulltext index (doesn't support more then one FTI per table)
+     *
+     * @param AbstractDb $collection
+     * @param string $indexTable
+     * @return array
+     */
+    protected function getFulltextIndexColumns(AbstractDb $collection, string $indexTable): array
+    {
+        $indexes = $collection->getConnection()->getIndexList($indexTable);
+        foreach ($indexes as $index) {
+            if (strtoupper($index['INDEX_TYPE']) === 'FULLTEXT') {
```

Comment:
Just a minor thing, I believe that using `\Magento\Framework\DB\Adapter\AdapterInterface::INDEX_TYPE_FULLTEXT` for comparison instead of the hardcoded value is a good idea.